src/main/java/com/sqlsheet/XlsResultSet.java

```diff
@@ -204,7 +204,16 @@ private Object getCellValue(int cellType, int columnType, Cell cell, int jdbcCol
 						+ XlsResultSetMetaData.columnTypeNameMap.get(columnType) + ".");
 			}
 		case Cell.CELL_TYPE_FORMULA:
-			return getCellValue(cell.getCachedFormulaResultType(), columnType, cell, jdbcColumn);
+			try {
+				return getCellValue(cell.getCachedFormulaResultType(), columnType, cell, jdbcColumn);
+			} catch (RuntimeException e) {
+				int formulaCellType = workbook.getCreationHelper().createFormulaEvaluator().evaluateFormulaCell(cell);
+				try {
+					return getCellValue(formulaCellType, columnType, cell, jdbcColumn);
+				} catch (Exception e2) {
+					throw e;
```

Comment:
You are catching e2 but throwing e. Why?

Reply:
coz in normaly situation xls has cached value, and it's primary error. but i want to try to evaluate formula, if it's not wokr - ok, forget it. And exceptions aren't so different in this case.